src/pybind/mgr/nfs/cluster.py

```diff
@@ -16,6 +15,19 @@
 log = logging.getLogger(__name__)
 
 
+def resolve_ip(hostname: str) -> str:
+    try:
+        r = socket.getaddrinfo(hostname, None, flags=socket.AI_CANONNAME,
+                               type=socket.SOCK_STREAM)
+        # pick first v4 IP, if present
+        for a in r:
+            if a[0] == socket.AF_INET:
+                return a[4][0]
+        return r[0][4][0]
+    except socket.gaierror as e:
+        raise NFSInvalidOperation(f"Cannot resolve IP for host {hostname}: {e}")
```

Comment:
move it to mgr_util.py?

Reply:
I tried that the first time around but (1) the exception is orchestrator-specific and generates a nice error message; I didn't want to adjust all of the catchers, and (2) i was hitting some issue with the tests I didn't want to debug.

I would rather remove this from mgr/nfs entirely... I'm not entirely sure we need it there at all:
```
                    if cluster.ip:
                        ip = cluster.ip
                    else:
                        c = self.mgr.get_hosts()
                        orchestrator.raise_if_exception(c)
                        hosts = [h for h in c.result
                                 if h.hostname == cluster.hostname]
                        if hosts:
                            ip = resolve_ip(hosts[0].addr)
                        else:
                            # sigh
                            ip = resolve_ip(cluster.hostname)
```